src/consensus/consensus.md

```diff
@@ -0,0 +1,293 @@
+chain at the intersection between the candidates, leveraging the property of
+Ouroboros that the honest chain will have more blocks than any other adversarial
+chain in a specific window of slots from the intersection point.
+
+Once the node finishes syncing with the network, this rule gracefully converges
+into the usual length-based comparison used in Praos, so the node can safely
+switch to running only Ouroboros Praos.
+
+### Chain diffusion MiniProtocols
+
+To fulfill the requirements above, the Consensus layer implements `ChainSync` and
+`BlockFetch`, miniprotocols to diffuse the blockchain through the network:
+
+| MiniProtocol | Purpose                                                                  |
+|:-------------|:-------------------------------------------------------------------------|
+| `ChainSync`  | Synchronize the changes on the nodes tips and transmit chains of headers |
+| `BlockFetch` | Transmit specific blocks                                                 |
+
+The specifications of the state machines of these miniprotocols are described in the
+Network documentation
+([design](https://ouroboros-network.cardano.intersectmbo.org/pdfs/network-design/network-design.pdf)
+and
+[spec](https://ouroboros-network.cardano.intersectmbo.org/pdfs/network-spec/network-spec.pdf)).
+
+## Transaction bookkeeping
+
+For blocks to have useful data, they have to contain transactions, which are codifications
+of operations on the Ledger state that only some authorized actors can enact. Notice that
+such transactions are what makes up the contents of the block. In one way, one could see
+the ledger as being able to validate transactions, and by implication full blocks as those
+are mainly collections of transactions.
+
+Note that although the nodes that will _use_ these transactions to create blocks are only
+the ones that forge new blocks, other nodes are mainly used to diffuse transactions to the
+majority of the network. It is expected that **all nodes** that participate in the network
+are able to receive, validate and distribute transactions.
+
+The Mempool is the abstract component that stores transactions which are valid on top of
+the latest known Ledger State. Being valid implies in particular that the transaction is
+pending, i.e. it has not been included in any prior block ever, as in Cardano every valid
+transaction must consume at least one UTxO.
+
+Usually pending transactions are discarded when the node is shut down, however this is not
+a hard requirement. It is conceivable for a node to store the received pending
+transactions on the disk and re-evaluate them on startup. As IOG's Haskell implementation
+does not store the pending transactions on the disk, the mempool is currently considered
+part of the Consensus layer, and not part of the Storage layer, as it does not store data
+on the disk.
+
+The `TxSubmission` miniprotocol is the one used to diffuse the pending transactions
```

Comment:
Perhaps nitpicking but is it not `TxSubmission2`?

Reply:
I think so, good point.

Reply:
The `2` is probably a thing for Network to discuss.

And a `pending transaction` is a transaction that is valid on the latest ledger state. If a transaction is pending it is in particular valid.